gettsim/benefits/wohngeld_dag.py

```diff
@@ -270,3 +274,42 @@ def _wohngeld_eink_abzüge_ab_2016(
         abzüge = pd.Series(dtype=float)
 
     return abzüge
+
+
+def _wohngeld_eink(
+    tu_id,
+    haushalts_größe,
+    _wohngeld_eink_abzüge,
+    _wohngeld_abzüge,
+    _wohngeld_brutto_eink,
+    _wohngeld_sonstiges_eink,
+    wohngeld_params,
+):
+    _wohngeld_eink_abzüge_per_tu = _wohngeld_eink_abzüge.groupby(tu_id).transform("sum")
+
+    vorläufiges_eink = (1 - _wohngeld_abzüge) * (
+        _wohngeld_brutto_eink + _wohngeld_sonstiges_eink - _wohngeld_eink_abzüge_per_tu
+    )
+
+    unteres_eink = [
+        wohngeld_params["min_eink"][hh_size]
+        if hh_size < 12
+        else wohngeld_params["min_eink"][12]
+        for hh_size in haushalts_größe
+    ]
+
+    return vorläufiges_eink.clip(lower=unteres_eink)
+
+
+def haushalts_größe(hh_id):
+    return hh_id.groupby(hh_id).transform("size")
+
+
+def _wohngeld_min_miete(haushalts_größe, wohngeld_params):
+    data = [
```

Comment:
Untested, but clearer what is happening (only to me?), no hard-coded 12, and vectorised:
```suggestion
df = haushaltsgröße.to_frame()
df["size_capped"] = df["haushaltsgröße"]
cap = wohngeld_params["min_miete"].index.max
df.loc[df["haushaltsgröße"] > cap, "size_capped"] = cap
out = pd.merge(
    left=haushaltsgröße,
    right=wohngeld_params["min_miete"],
    left_on="size_capped",
    right_index=True
)["min_miete"]
```

Reply:
I will fix the hardcoded 12. The merge operation does not pay of for 10,000 individuals, but 100,000 it only needs halve the time (15ms vs 34). And clipping at 12 and then `s.replace` is even faster with 4ms. I always though it is slow for some reason.

Just in general, should we try to express most operations as pandas operations because cudf knows how to handle them efficiently and mixed operations will produce calculations on the CPU and then cudf on the GPU with a lot of data copying?

Reply:
The standard pandas operations should be okay, I guess we'll do something like a conditional import at the top eventually or write a little wrapper.